src/Service/TransactionService.php

```diff
@@ -243,34 +246,58 @@ public function processTransaction($transaction)
         return $apiPayment;
     }
 
-    /**
-     * Retrieves the OrderPayment object, created at validateOrder. And add transaction data.
-     *
-     * @param string $molliePaymentId
-     * @param string $molliePaymentMethod
-     * @param int $orderId
-     *
-     * @return void
-     * @throws PrestaShopDatabaseException
-     * @throws PrestaShopException
-     */
-    protected function saveOrderTransactionData($molliePaymentId, $molliePaymentMethod, $orderId)
+    public function updateOrderTransaction($transactionId, $orderReference)
     {
-        // retrieve ALL payments of order.
-        // in the case of a cancel or expired on banktransfer, this will fire too.
-        // if no OrderPayment objects is retrieved in the collection, do nothing.
-        $order = new Order((int)$orderId);
-        $collection = OrderPayment::getByOrderReference($order->reference);
-        if (count($collection) > 0) {
-            /** @var OrderPayment $orderPayment */
-            $orderPayment = $collection[0];
-
-            // for older versions (1.5) , we check if it hasn't been filled yet.
-            if (!$orderPayment->transaction_id) {
-                $orderPayment->transaction_id = $molliePaymentId;
-                $orderPayment->payment_method = $molliePaymentMethod;
-                $orderPayment->update();
+        $transactionInfos = [];
```

Comment:
I surely recall I mentioned how I dont like nested foreach statements + many conditionals inside of the foreach. There are patterns like "Pipeline" which would handle such cases - but I see an efortless try is to at the very least divide into many private functions then you maybe some parts could be even utilities for improved testability.